src/nrnpython/nrnpy_hoc.cpp

```diff
@@ -2064,6 +2092,58 @@ static int hocobj_setitem(PyObject* self, Py_ssize_t i, PyObject* arg) {
     return err;
 }
 
+static int hocobj_slice_setitem(PyObject* self, PyObject* slice, PyObject* arg) {
+    // Non slice indexing still uses original function
+    if (!PySlice_Check(slice)) {
+        return hocobj_setitem(self, PyLong_AsLong(slice), arg);
+    }
+    auto* po = (PyHocObject*) self;
+    if (!po->ho_) {
+        PyErr_SetString(PyExc_TypeError, "Obj is NULL");
+        return -1;
+    }
+    if (po->type_ != PyHoc::HocObject || po->ho_->ctemplate != hoc_vec_template_) {
+        PyErr_SetString(PyExc_TypeError, "sequence index must be integer, not 'slice'");
+        return -1;
+    }
+    Vect* v = (Vect*) po->ho_->u.this_pointer;
+    Py_ssize_t start = 0;
+    Py_ssize_t end = 0;
+    Py_ssize_t step = 0;
+    Py_ssize_t slicelen = 0;
+    Py_ssize_t cap = vector_capacity(v);
+    PySlice_GetIndicesEx(slice, cap, &start, &end, &step, &slicelen);
+    // Slice index assignment requires a list of the same size as the slice
+    PyObject* iter = PyObject_GetIter(arg);
+    if (!iter) {
+        PyErr_SetString(PyExc_TypeError, "can only assign an iterable");
+        return -1;
+    }
+    PyObject* val = NULL;
+    for (Py_ssize_t i = 0; i < slicelen; ++i) {
+        val = PyIter_Next(iter);
+        if (!val) {
+            Py_DECREF(iter);
+            PyErr_SetString(PyExc_IndexError, "iterable object must have the same length as slice");
+            return -1;
+        }
+        PyArg_Parse(val, "d", vector_vec(v) + (i * step + start));
+        Py_DECREF(val);
+    }
+    val = PyIter_Next(iter);
+    if (val) {
+        while (val) {
+            Py_DECREF(val);
+            val = PyIter_Next(iter);
```

Comment:
Don't need a `while` here. No matter what happens, we know the iterable object is too long, no point iterating it through.

Also, not all iterables have finite length; consider:
```
def my_iter():
    while True:
        yield 1

for item in my_iter():
    print(item)
    # this runs forever; ctrl-C when bored
```